stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/nativeimpl/ParseHeaderValue.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.net.http.nativeimpl;
+
+import org.ballerinalang.bre.Context;
+import org.ballerinalang.model.types.TypeKind;
+import org.ballerinalang.model.values.BMap;
+import org.ballerinalang.model.values.BString;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.natives.AbstractNativeFunction;
+import org.ballerinalang.natives.annotations.Argument;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.ReturnType;
+import org.ballerinalang.net.http.HttpUtil;
+import org.ballerinalang.util.exceptions.BLangNullReferenceException;
+import org.ballerinalang.util.exceptions.BallerinaException;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Native function to parse header value and get value with parameter map.
+ *
+ * @since 0.96.1
+ */
+@BallerinaFunction(
+        packageName = "ballerina.net.http",
+        functionName = "parseHeaderValue",
+        args = {@Argument(name = "headerValue", type = TypeKind.STRING)},
+        returnType = {@ReturnType(type = TypeKind.STRING),
+                @ReturnType(type = TypeKind.MAP, elementType = TypeKind.STRING)},
+        isPublic = true
+)
+public class ParseHeaderValue extends AbstractNativeFunction {
+
+    private static final String SEMICOLON = ";";
+    private static final String COMMA = ",";
+    private static final String PARSER_ERROR = "failed to parse: ";
+
+    @Override
+    public BValue[] execute(Context context) {
+        try {
+            String headerValue = getStringArgument(context, 0);
+            if (headerValue.contains(COMMA)) {
+                headerValue = headerValue.substring(0, headerValue.indexOf(COMMA));
+            }
+            return getValueAndParamMap(headerValue);
+        } catch (BLangNullReferenceException ex) {
+            throw new BallerinaException(PARSER_ERROR + "header value cannot be null");
```

Comment:
Shouldn't we return this type of errors ?

Reply:
Changed from throwing errors to return errors